lib/SimpleSAML/Utils/HTTP.php

```diff
@@ -1165,7 +1165,7 @@ public static function setCookie($name, $value, $params = null, $throw = true)
             'lifetime' => 0,
             'expire'   => null,
             'path'     => '/',
-            'domain'   => null,
+            'domain'   => version_compare(PHP_VERSION, '7.3.0', '>=') ? null : '',
```

Comment:
It has to do with the different signatures for `setcookie` and `setrawcookie` pre/post PHP 7.3..
Pre 7.3 we need a string, post we need $options['domain'].. We used to set it to `null` for pre 7.3 as well and that's an issue with strict_types.. This happened to be the easiest fix

Reply:
I see...

From what I understand from the documentation, the options array accepts keys with values the same as the parameters in the original function's signature. Would it then be possible to just set it to `' '`, regardless the PHP version we are running?

Reply:
Maybe I'm missing the point,  but the options-array wasn't added until PHP 7.3?

Reply:
Yes, but I'm not talking about using that signature unconditionally, but about this specific change setting the `domain` to `null` or an empty string depending on the PHP version. Maybe I'm misinterpreting your change, but the outcome here is that for PHP 7.3 or later, you will call `setcookie()` or `setrawcookie()` with an `$options` array where the default value for `'domain'` is `null`. If it's an older version of PHP, we're passing an empty string to the `$domain` parameter.

Reply:
Ah right, I got what you mean now..  The empty string will cause an implicit cast to null, which is going to bug us again if we merge the strict_types=1 PR..

Reply:
Uhm, why should it implicitly cast to `null`? I think I'm missing something, because setting an element of an array to an empty string will (and *should*) result in that element being a string. Empty, but a string:

```php
php > $a = [ 'b' => '' ];
php > var_dump($a);
array(1) {
  ["b"]=>
  string(0) ""
}
```

Reply:
Nevermind, I'm an idiot..  Took care of it!